improver_tests/utilities/copy_attributes/test_CopyAttributes.py

```diff
@@ -0,0 +1,73 @@
+# (C) Crown copyright, Met Office. All rights reserved.
+#
+# This file is part of improver and is released under a BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+from unittest.mock import patch, sentinel
+
+from iris.cube import Cube
+
+from improver.utilities.copy_attributes import CopyAttributes
+
+
+class HaltExecution(Exception):
+    pass
+
+
+@patch("improver.utilities.copy_attributes.as_cube")
+@patch("improver.utilities.copy_attributes.as_cubelist")
+def test_as_cubelist_called(mock_as_cubelist, mock_as_cube):
+    mock_as_cubelist.return_value = sentinel.cubelist
+    mock_as_cube.side_effect = HaltExecution
+    try:
+        CopyAttributes(["attribA", "attribB"])(
+            sentinel.cube0, sentinel.cube1, template_cube=sentinel.template_cube
+        )
+    except HaltExecution:
+        pass
+    mock_as_cubelist.assert_called_once_with(sentinel.cube0, sentinel.cube1)
+    mock_as_cube.assert_called_once_with(sentinel.template_cube)
+
+
+def test_copy_attributes_multi_input():
+    """
+    Test the copy_attributes function for multiple input cubes.
+
+    Demonstrates copying attributes from the template cube to the input
+    cubes and also demonstrates the attributes on the templates cube that
+    aren't specified in the attributes list awre indeed ignored.
+    """
+    attributes = ["attribA", "attribB"]
+    cube0 = Cube([0], attributes={"attribA": "valueA", "attribB": "valueB"})
+    cube1 = Cube([0], attributes={"attribA": "valueAA", "attribB": "valueBB"})
+    template_cube = Cube(
+        [0], attributes={"attribA": "tempA", "attribB": "tempB", "attribC": "tempC"}
+    )
+
+    plugin = CopyAttributes(attributes)
+    result = plugin.process(cube0, cube1, template_cube=template_cube)
+    assert type(result) == tuple
+    for res in result:
+        assert res.attributes["attribA"] == "tempA"
+        assert res.attributes["attribB"] == "tempB"
+        assert "attribC" not in res.attributes
+    assert id(result[0]) == id(cube0)
+    assert id(result[1]) == id(cube1)
```

Comment:
Note how we are verifying the object IDs, since `CubeAttributes` is an in-place operation.